acra/acra-in-depth/data-flow/_index.md

```diff
@@ -1,4 +1,113 @@
+
+- [_AcraServer_]({{< ref "/acra/configuring-maintaining/general-configuration/acra_server.md" >}}) — a separate daemon that runs in an isolated environment (separate virtual machine or physical server). AcraServer is responsible for holding all the secrets required to decrypt the data and for actually decrypting this data.
+
+- [_AcraCensor_]({{< ref "/acra/security-controls/sql-firewall/" >}}) – is a firewall-like component that sits inside AcraServer, and checks every SQL query to the database.
+
+  ![](/files/data-flow/acra-archi-tls.png)
+  
+  _Acra-based application architecture using TLS transport encryption._
+  
+Second way to protect raw data between application and AcraServer/AcraTranslator is to use AcraConnector:
+
+- [_AcraConnector_]({{< ref "/acra/configuring-maintaining/general-configuration/acra_connector.md" >}}) — a client-side daemon that runs under a separate user / in a separate container, and which acts as a database listener that redirects all the queries to AcraServer and, upon receiving the results, feeds it back into the app. AcraConnector is an optional component, required in systems that are using extra transport encryption layer via Themis SecureSession.
+
+  ![](/files/data-flow/acra-entities.png)    
+  
+  _Acra-based application architecture using TLS transport encryption._
+  
+### Data flow schemas
+
+With TLS as secure transport between application and AcraServer/AcraTranslator:
+1. **Writing**: Application --> AcraTranslator --> Application --> Storage.
+
+   **Description**: application uses AcraTranslator to encrypt/decrypt data and save it in a storage. 
+  
+   **Reading**: Application --> Storage --> Application --> AcraTranslator --> Application.
+
+   **Description**: application uses AcraTranslator to encrypt/decrypt data and save it in a storage.
+
+2. **Writing**: Application --> AcraWriter --> Database.
+
+   **Description**: application uses AcraWriter and public key to encrypt data and save it directly to database.
+
+   **Reading**: Database --> AcraServer --> Application.
+
+   **Description**: application request data from database through AcraServer that decrypts it transparently.
+  
+3. **Writing**: Application --> AcraServer --> Database.
+
+   **Description**: application sends data to database through AcraServer in [Transparent encryption mode](/acra/configuring-maintaining/general-configuration/acra-server/#transparent-proxy-mode-INVALID).
+   All data manipulations like [encryption]({{< ref "/acra/security-controls/encryption/" >}})/
+   [tokenization]({{< ref "/acra/security-controls/tokenisation/" >}})/[masking]({{< ref "/acra/security-controls/masking/" >}})
+   applied on AcraServer side before sending to database. 
+
+   **Reading**: Database --> AcraServer --> Application.
+
+   **Description**: application request data from database through AcraServer. AcraServer transparently
+   decrypts/detokinze/unmask data and pass it to application.
```

Comment:
```suggestion
   decrypts/detokinzes/unmasks data and passes it to application.
```

Reply:
applies it manually because of `detokenizes`, not `detokinzes` ))